libbroker/broker/envelope.cc

```diff
@@ -148,6 +152,39 @@ expected<envelope_ptr> envelope::deserialize(const std::byte* data,
   }
 }
 
+expected<envelope_ptr> envelope::deserialize_json(const char* data,
+                                                  size_t size) {
+  // Parse the JSON text into a JSON object.
+  auto val = caf::json_value::parse_shallow(std::string_view{data, size});
+  if (!val)
+    return error{ec::invalid_json};
+  auto obj = val->to_object();
+  // Type-checking.
+  if (obj.value("type").to_string() != "data-message")
```

Comment:
Hmm, I realize I'm looking at parsing one layer further in: The idea was, JSON contains only the serialized event vector without the `data-message` and `topic`.

In my use-case, "type" and "topic"  are in message headers, not within the payload and the payload isn't meant to be structured like the websocket protocol.

Let me check if the JSON serialization in #391 even does what I thought it did.

Reply:
So you'd basically want to have a function that parses you just the `variant` from the JSON format? I'll add that later.

Reply:
> Now, the main motivation is to "just" re-use existing encoding/serialization functionality from broker for events. If you think that API doesn't make much sense to have, feel free to push back.

No push back at all, I just didn't think of that use case yet. 🙂

I'll add functions that do the JSON (de)-serialization for `variant` as well.